### PR TITLE
Change Libcurl option from "winssl" to "schannel"

### PR DESCRIPTION
**Pull request by stefan-floeren**
Specify library name and version:  **libcurl/***

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

The winssl option was removed in version 7.72.0. The new (schannel)
options are supported since 7.64.0, so they can be used for all available
versions.

Addresses the windows portion of #2887.

### CODE REVIEW

File: recipes/libcurl/all/conanfile.py [hunk 1]
```diff
@@ -92,12 +95,16 @@ def configure(self):
...
-        # - with_openssl AND with_winssl raises to error
-        # - with_openssl AND NOT with_winssl uses openssl
-        # Moreover darwin_ssl is set by default and with_winssl is not
+        # - with_openssl AND with_schannel raises to error
+        # - with_openssl AND NOT with_schannel uses openssl
+        # Moreover darwin_ssl is set by default and with_schannel is not
+
+        if self.settings.os == "Windows" and self.options.with_winssl:
+            self.output.warn("with_winssl option is deprecated, use with_schannel option instead.")
+            self.options.with_schannel = True
```
Comment (madebr) [inline recipes/libcurl/all/conanfile.py:104]:
```suggestion
        if self.settings.os == "Windows" and self.options.with_winssl:
            self.output.warn("with_winssl option is deprecated, use with_schannel option instead.")
            self.options.with_schannel = True
        del self.options.with_winssl
```

**Comment by madebr** [comment]
@SpaceIm is doing a major refactor of the ssl options in #2880
This work will surely create conflicts.
It would be great if these efforts would get combined earlier rather then late.

**Comment by stefan-floeren** [comment]
I'm not opposed to dropping this PR if we add this to the rework. I can't push to the other PR, so we need to wait for @SpaceIm to add it.

I'll change this to draft and will adjust it after the other is merged / close if integrated.

**Comment by SpaceIm** [comment]
Yep, I can incorporate this modification in https://github.com/conan-io/conan-center-index/pull/2880, it will avoid to deprecate a value in the new option `with_ssl`.

Something like this:

```patch
--- a/recipes/libcurl/all/conanfile.py
+++ b/recipes/libcurl/all/conanfile.py
@@ -19,7 +19,7 @@ class LibcurlConan(ConanFile):
     settings = "os", "arch", "compiler", "build_type"
     options = {"shared": [True, False],
                "fPIC": [True, False],
-               "with_ssl": [False, "openssl", "wolfssl", "winssl", "darwinssl"],
+               "with_ssl": [False, "openssl", "wolfssl", "schannel", "darwinssl"],
                "with_openssl": [True, False],
                "with_wolfssl": [True, False],
                "with_winssl": [True, False],
@@ -115,7 +115,7 @@ class LibcurlConan(ConanFile):
                 self.options.with_ssl = False
         if not tools.is_apple_os(self.settings.os) and self.options.with_ssl == "openssl":
             if self.settings.os == "Windows" and self.options.with_winssl:
-                self.options.with_ssl = "winssl"
+                self.options.with_ssl = "schannel"
             elif self.options.with_openssl:
                 self.options.with_ssl = "openssl"
             elif self.options.with_wolfssl:
@@ -128,8 +128,8 @@ class LibcurlConan(ConanFile):
         del self.options.with_wolfssl
         # ===============================
 
-        if self.options.with_ssl == "winssl" and self.settings.os != "Windows":
-            raise ConanInvalidConfiguration("winssl only suppported on Windows.")
+        if self.options.with_ssl == "schannel" and self.settings.os != "Windows":
+            raise ConanInvalidConfiguration("schannel only suppported on Windows.")
         if self.options.with_ssl == "darwinssl" and not tools.is_apple_os(self.settings.os):
             raise ConanInvalidConfiguration("darwinssl only suppported on Apple like OS (Macos, iOS, watchOS or tvOS).")
         if self.options.with_ssl == "wolfssl" and self._is_using_cmake_build and tools.Version(self.version) < "7.70.0":
@@ -222,7 +222,7 @@ class LibcurlConan(ConanFile):
             wolfssl_path = self.deps_cpp_info["wolfssl"].rootpath.replace("\\", "/")
             params.append("--with-wolfssl=%s" % wolfssl_path)
             params.append("--without-ssl")
-        elif self.options.with_ssl == "winssl":
+        elif self.options.with_ssl == "schannel":
             params.append("--with-winssl")
             params.append("--without-ssl")
         elif self.options.with_ssl == "darwinssl":
@@ -467,7 +467,7 @@ class LibcurlConan(ConanFile):
         self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
         self._cmake.definitions["CURL_STATICLIB"] = not self.options.shared
         self._cmake.definitions["CMAKE_DEBUG_POSTFIX"] = ""
-        self._cmake.definitions["CMAKE_USE_WINSSL"] = self.options.with_ssl == "winssl"
+        self._cmake.definitions["CMAKE_USE_SCHANNEL"] = self.options.with_ssl == "schannel"
         self._cmake.definitions["CMAKE_USE_OPENSSL"] = self.options.with_ssl == "openssl"
         if tools.Version(self.version) >= "7.70.0":
             self._cmake.definitions["CMAKE_USE_WOLFSSL"] = self.options.with_ssl == "wolfssl"
@@ -536,7 +536,7 @@ class LibcurlConan(ConanFile):
             self.cpp_info.components["curl"].system_libs = ["ws2_32"]
             if self.options.with_ldap:
                 self.cpp_info.components["curl"].system_libs.append("wldap32")
-            if self.options.with_ssl == "winssl":
+            if self.options.with_ssl == "schannel":
                 self.cpp_info.components["curl"].system_libs.append("Crypt32")
         elif self.settings.os == "Macos":
             if self.options.with_ldap:
```

I can check, but do you now if `configure.ac` has also introduced a new option like `--with-schannel` ?

**Comment by madebr** [comment]
@SpaceIm 
Current master's configure.ac looks like they have the same effect:
https://github.com/curl/curl/blob/11ab0ad60f1e106ad84f33ea9843c0da0433e49e/configure.ac#L1613-L1622
They both set `OPT_SCHANNEL`

**Comment by SpaceIm** [comment]
Ok and I also see `with-schannel` option in the oldest libcurl version in CCI.

**Comment by SpaceIm** [comment]
Commit added in #2880: https://github.com/conan-io/conan-center-index/pull/2880/commits/7024ccbd15e0e43ad7a3d1172a51e78d94062114

**Comment by stefan-floeren** [comment]
Looks good.

**Pull request closed without merging**